flow/shared/exceptions/postgres.go

```diff
@@ -0,0 +1,14 @@
+package exceptions
+
+// PostgresSetupError represents errors during setup of Postgres peers, maybe we can later replace with a more generic error type
+type PostgresSetupError struct {
```

Comment:
should implement `Unwrap() error` to work better with various functions from `errors`

Reply:
> I don't see what this adds over fmt.Errorf

This is just for easier check, so we don't have to do string matching, have used it in 1 place and can gradually increase usage

Reply:
ideal would be for there to be interface that has `ErrorClass() ErrorClass`

Reply:
agreed, I think we can do that once we have more error types